vcr/holder/oidc4vci_wallet_test.go

```diff
@@ -126,6 +129,40 @@ func Test_wallet_HandleCredentialOffer(t *testing.T) {
 			return vcStored.Load() != nil, nil
 		}, 2*time.Second, "time-out waiting for VC to be stored")
 	})
+	t.Run("pre-authorized code grant", func(t *testing.T) {
+		w := NewOIDCWallet(holderDID, "https://holder.example.com", nil, nil, nil, time.Second*5).(*wallet)
+		t.Run("no grants", func(t *testing.T) {
+			offer := oidc4vci.CredentialOffer{Credentials: []map[string]interface{}{{}}}
+			err := w.HandleCredentialOffer(audit.TestContext(), offer)
+			require.EqualError(t, err, "invalid_grant - couldn't find (valid) pre-authorized code grant in credential offer")
+		})
+		t.Run("no pre-authorized grant", func(t *testing.T) {
+			offer := oidc4vci.CredentialOffer{
+				Credentials: []map[string]interface{}{{}},
```

Comment:
Is there any way to simplify this? Maybe defining a type or two above? This looks like something from a golang proficiency exam

Reply:
Wrapped it in a function with docs.